cloud-gateway-service/src/test/java/org/zowe/apiml/cloudgatewayservice/service/RouteLocatorTest.java

```diff
@@ -10,78 +10,232 @@
+
+    private static SchemeHandler createSchemeHandler(AuthenticationScheme type) {
+        SchemeHandler out = mock(SchemeHandler.class);
+        doReturn(type).when(out).getAuthenticationScheme();
+        return out;
     }
 
-    ServiceInstance instance = new DefaultServiceInstance("gateway-10012", "gateway", "gatewayhost", 10012, true, metadata);
-    ServiceInstance instance2 = new DefaultServiceInstance("gateway-2-10012", "gateway", "gatewayhost-2", 10012, true, metadata);
-    ReactiveDiscoveryClient dc = mock(ReactiveDiscoveryClient.class);
-    DiscoveryLocatorProperties properties = new DiscoveryLocatorProperties();
+    private static RouteDefinitionProducer createRouteDefinitionProducer(int order, String id) {
+        EurekaMetadataParser metadataParser = new EurekaMetadataParser();
+        RouteDefinitionProducer rdp = mock(RouteDefinitionProducer.class);
+        doReturn(order).when(rdp).getOrder();
+        doAnswer(answer -> {
+            ServiceInstance serviceInstance = answer.getArgument(0);
+            RoutedService routedService = answer.getArgument(1);
+
+            RouteDefinition routeDefinition = new RouteDefinition();
+            Map<String, Object> metadata = new HashMap<>();
+            metadata.put("serviceId", serviceInstance.getServiceId());
+            metadata.put("gatewayUrl", routedService.getGatewayUrl());
+            metadata.put("producerId", id);
+            routeDefinition.setMetadata(metadata);
+
+            return routeDefinition;
+        }).when(rdp).get(any(), any());
+        return rdp;
+    }
 
     @Nested
-    class GivenRouteLocator {
+    class CommonMethods {
+
+        @Test
+        void givenDiscoveryClient_whenGetServiceInstances_thenMapThem() {
+            when(discoveryClient.getServices()).thenReturn(Flux.fromArray(new String[] {"service1", "service2"}));
+            ServiceInstance serviceInstance1 = mock(ServiceInstance.class);
+            ServiceInstance serviceInstance2 = mock(ServiceInstance.class);
+            doReturn(Flux.just(serviceInstance1)).when(discoveryClient).getInstances("service1");
+            doReturn(Flux.just(serviceInstance2)).when(discoveryClient).getInstances("service2");
+
+            doCallRealMethod().when(routeLocator).getServiceInstances();
+            assertArrayEquals(
+                new Object[] {Collections.singletonList(serviceInstance1), Collections.singletonList(serviceInstance2)},
+                routeLocator.getServiceInstances().toStream().toArray()
+            );
+        }
+
+        @Test
+        void givenNoAuthentication_whenSetAuth_thenDoNothing() {
+            assertDoesNotThrow(() -> routeLocator.setAuth(null, null));
+        }
+
+        @Test
+        void givenNoAuthenticationScheme_whenSetAuth_thenDoNothing() {
+            assertDoesNotThrow(() -> routeLocator.setAuth(null, new Authentication()));
+        }
+
+        @Test
+        void givenUnknownAuthenticationScheme_whenSetAuth_thenDoNothing() {
```

Comment:
why is x509 the unknown scheme?

Reply:
Because in the mock implementation is not registred any filter, I have changed name of test to givenAuthenticationSchemeWithoutFilter_whenSetAuth_thenDoNothing